FraudProofs.md

```diff
@@ -0,0 +1,158 @@
+false, invalid, or conflicting attestations, their fraud should be easily and compactly provable
+and publicly verifiable.
+After all, oracle fraud comes in the form of an unforgeable attestation of false or invalid events,
+or else in the form of multiple conflicting unforgeable attestations.
+
+This document specifies a standard format for serializing fraud proofs as well as specifying their verification.
+
+## Table of Contents
+
+* [Proof of Oracle Attestation](#proof-of-oracle-attestation)
+  * [Version 0 `oracle_attestation_proof`](#version-0-oracle_attestation_proof)
+* [Claim of False Oracle Attestation](#claim-of-false-oracle-attestation)
+  * [Version 0 `false_oracle_attestation_claim` ](#version-0-false_oracle_attestation_claim)
+* [Proof of Invalid Oracle Attestation](#proof-of-invalid-oracle-attestation)
+  * [Version 0 `invalid_oracle_attestation_proof`](#version-0-invalid_oracle_attestation_proof)
+* [Proof of Oracle Equivocation](#proof-of-oracle-equivocation)
+  * [Version 0 `oracle_equivocation_proof`](#version-0-oracle_equivocation_proof)
+  * [Version 1 `oracle_equivocation_proof`](#version-1-oracle_equivocation_proof)
+* [Authors](#authors)
+
+## Proof of Oracle Attestation
+
+The fundamental building block of DLC oracle fraud proofs is a proof that a specific oracle
+attested to a specific outcome, which can then be used with other information to prove fraud.
+
+In theory, it would be best if DLC oracles could deliver their announcements and attestations
+using some form of broadcast channel, but in practice it is expected that oracle messages will
+often times be received by users via APIs and other more private and P2P means.
+
+If DLC oracle broadcast channels did exist in such a way that oracles could not commit fraud privately,
+then oracle attestation proofs would be as simple as wrapping the publicly broadcast oracle messages,
+but practically speaking oracles have the ability to cheat without broadcasting fraudulent attestations publicly.
+
+Luckily, enough information is leaked to cheated DLC participants on-chain that they can still
+always generate proofs which show that a fraudulent oracle attestation must exist.
+
+#### Version 0 `oracle_attestation_proof`
+
+1. type: ??? (`oracle_attestation_proof_v0`)
+2. data:
+   * [`u16`:`num_oracles`]
+   * [`num_oracles*oracle_announcement`:`oracle_announcements`]
+   * [`num_oracles*oracle_outcome`:`oracle_outcomes`]
+   * [`32*bytes`:`aggregate_oracle_attestation`]
+   * [`u16`:`oracle_index`]
+3. subtype: `oracle_outcome`
+4. data:
+   * [`nb_signatures*string`:`outcomes`]
+
+The most important piece of this proof is the `aggregate_oracle_attestation` which is recoverable from
+on-chain information as the difference between the broadcast CET's signature and its corresponding adaptor signature.
+In the case that one has access directly to the `oracle_attestation` in question, then this can be used in place of an aggregate.
```

Comment:
It's not clear to me if you're referring to attestation (which in the specs refer to a message) or a signature. Given the type  assume the latter but then better to distinguish names I feel. Also how do you know which one it is (aggregate or not)?

Reply:
changed it to `In the case that one has access directly to an oracle's attestation, then this can be used as the aggregate.` (because it is just an aggregate of one).